source/adapters/level_zero/command_buffer.cpp

```diff
@@ -22,6 +22,71 @@
 
 namespace {
 
+// Checks whether zeCommandListImmediateAppendCommandListsExp can be used for a
+// given Context and Device.
+bool checkImmediateAppendSupport(ur_context_handle_t Context,
+                                 ur_device_handle_t Device) {
+  // TODO The L0 driver is not reporting this extension yet. Once it does,
+  // switch to using the variable zeDriverImmediateCommandListAppendFound.
+
+  // Minimum version that supports zeCommandListImmediateAppendCommandListsExp.
+  constexpr uint32_t MinDriverVersion = 30898;
+  bool DriverSupportsImmediateAppend =
+      Context->getPlatform()->isDriverVersionNewerOrSimilar(1, 3,
+                                                            MinDriverVersion);
+
+  // If this environment variable is:
+  //   - Set to 1: the immediate append path will always be enabled as long the
+  //   pre-requisites are met.
+  //   - Set to 0: the immediate append path will always be disabled.
+  //   - Not Defined: The default behaviour will be used which enables the
+  //   immediate append path only for some devices when the pre-requisites are
+  //   met.
+  const char *AppendEnvVarName = "UR_L0_CMD_BUFFER_USE_IMMEDIATE_APPEND_PATH";
```

Comment:
We should document this in https://github.com/oneapi-src/unified-runtime/blob/main/scripts/core/LEVEL_ZERO.rst